xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/src/main/java/org/xwiki/notifications/sources/internal/DefaultNotificationParametersFactory.java

```diff
@@ -393,6 +395,9 @@ public void useUserPreferences(NotificationParameters parameters) throws Notific
                 enableAllEventTypes(parameters);
 
                 parameters.filters.add(new ForUserEventFilter(parameters.format, null));
+                // FIXME: This doesn't feel right: we actually also retrieve the filters for pre-filtering.
```

Comment:
We are back to what I was talking about in the chat: the fact that IMO `getAllFilters` should check the configured mode to know if the post filterinig phase is supposed to use all filters supporting post filtering or only filters supporting post filtering. That way you can remove the `if` on the mode in `useUserPreferences`.

Reply:
> if the post filterinig phase is supposed to use all filters supporting post filtering or only filters supporting post filtering

The problem is that such check is not made possible with NotificationFilter#isPhaseSupported

Reply:
Yes and no, since we have only two phases right now you would invert it (check if it's not supporting PREFILTERING when you want only POSTFILTERING) but it indeed won't work if we have more than two phase and it's a bit hackish.

Reply:
> I don't think moving the configuration check to the filter would be a good idea. We might need to know what a filter support whatever the current configuration.

I agree on that one.

> I guess we could have both a NotificationFilter#isPhaseSupported and NotificationFilter#isOnlyPhaseSupported and choose one of the other depending on the configuration.

Not a big fan of the second one. Sounds like I could reintroduce `NotificationFilter#getSupportedPhase` and even use it in most cases for `NotificationFilter#isPhaseSupported`. It would probably mean reintroducing BOTH phase too. I think it's better like that.

Reply:
I think NotificationFilter#getSupportedPhases would make more sense. That way you keep the enum and getAllFilters as it is.